docs/docs/adrs/adr-018-permissionless-ics.md

```diff
@@ -0,0 +1,266 @@
+| `ConsumerValidatorBytePrefix`            |Stores consumer validators for a specific chain                                   |**YES**                   |Potentially at every [epoch](ADR on epochs)                               |
+| `OptedInBytePrefix`                      |Stores opted-in validators for a specific chain                                   |**YES**                   |Potentially at every block                                                |
+| `TopNBytePrefix`                         |Stores whether a consumer chain is Top N or not                                   |**YES**                   |Every parameter update                                                    |
+| `ValidatorsPowerCapPrefix`               |Stores ther power cap of a chain                                                  |**YES**                   |Every parameter update                                                    |
+| `ValidatorSetCapPrefix`                  |Stores the set cap of a chain                                                     |**YES**                   |Every parameter update                                                    |
+| `AllowlistPrefix`                        |Stores the allowlist of a chain                                                   |**YES**                   |Every parameter update                                                    |
+| `DenylistPrefix`                         |Stores the denylist of a chain                                                    |**YES**                   |Every parameter update                                                    |
+| `ConsumerRewardsAllocationBytePrefix`    |Stores the ICS rewards per chain                                                  |**YES**                   |Every IBC transfer packet that sends rewards to the provider              |
+| `ConsumerCommissionRatePrefix`           |Comission rate per chain per validator                                            |**YES**                   |Every `MsgSetConsumerCommissionRate` message                              |
+| `MinimumPowerInTopNBytePrefix`           |Stores the minimum power needed to opt in for a chain                             |**YES**                   |Every epoch                                                               |
+| `ConsumerAddrsToPruneV2BytePrefix`       |Stores consumer addresses to be pruned (as part of `VSCMaturedPacket`s deprecation)|**YES**                   |Every `MsgAssignConsumerKey` or `MsgOptIn` and later during actual pruning|
+
+Everything stored under a key associated with a `chainID` needs to be migrated to new state under `consumerID`. 
+Because we have to migrate in any case, we can also clean up a number of those keys by building a `ConsumerChainRecord`.
```

Comment:
I'm still not a huge fan of `ConsumerChainRecord`. It seems to slightly complicate things (e.g., someone adding new state now would have to think on whether this should go in `ConsumerChainRecord` or not) without any extra benefits because it does not contain everything currently stored under `chainID`-related keys.

I wonder if it would make sense to have a Go `struct` that is a `ConsumerChainRecord` and provides methods to do everything you want to do for a chain (e.g., `chainRecord.SetConsumerValidator(validator)`, `chainRecord.GetConsumerValidators()`, etc.). Those methods under the hood would be using the `proto`-generated `structs` and this way we won't have to do any migration besides that of Neutron and Stride.

Reply:
Resolving this as outdated. `ConsumerChainRecord` or any substantial refactoring on keys is orthogonal to permissionless. I do not believe it should be part of this ADR.